geniushubclient/zone.py

```diff
@@ -39,9 +38,20 @@ class GeniusZone(GeniusBase):
     def __init__(self, zone_id, raw_json, hub) -> None:
         super().__init__(zone_id, raw_json, hub, ATTRS_ZONE)
 
+        self._properties = Properties()
+
         self.device_objs = []
         self.device_by_id = {}
 
+        try:
+            self.update(raw_json, self._hub.api_version)
+        except (AttributeError, LookupError, TypeError, ValueError):
+            _LOGGER.exception("Failed to convert Zone %s.", zone_id)
+
+    def update(self, json_data: Dict, api_version: int):
+        """Parse the json for the zone data."""
+        self._properties.update(json_data, api_version)
```

Comment:
For no other reason than that was where I first thought of putting it, and have given it no further thought since! You're right though, it should be in the update method.